src/DurableTask.SqlServer/SqlDbManager.cs

```diff
@@ -171,6 +173,57 @@ async Task<DatabaseLock> AcquireDatabaseLockAsync()
             return new DatabaseLock(connection, lockTransaction);
         }
 
+        async Task<SqlConnection> CreateDatabaseConnectionAsync()
+        {
+            if (!this.settings.CreateDatabaseIfNotExists)
+            {
+                return this.settings.CreateConnection();
+            }
+
+            // Note that we may not be able to connect to the DB, let alone obtain the lock,
+            // if the database does not exist yet. So we obtain a connection to the 'master' database for now.
+            SqlConnection connection = this.settings.CreateConnection("master");
+            await connection.OpenAsync();
+
+            if (!await this.DoesDatabaseExistAsync(this.settings.DatabaseName, connection))
+            {
+                await this.CreateDatabaseAsync(this.settings.DatabaseName, connection);
+            }
+
+#if NETSTANDARD2_0
```

Comment:
It's kind of ugly to bifurcate like this lol. I can use the same non-async overload for both frameworks too